packages/ui/src/services/mapping.service.ts

```diff
@@ -183,6 +183,24 @@ export class MappingService {
     wrapped.parent = wrapper;
   }
 
+  /**
+   * Renames a parameter in all mappings within the mapping tree.
+   * This updates XPath expressions that reference the old parameter name.
+   */
+  static renameParameterInMappings(
+    item: MappingTree | MappingItem,
+    oldDocumentId: string,
+    newDocumentId: string,
+  ): void {
+    item.children.forEach((child) => {
+      MappingService.renameParameterInMappings(child, oldDocumentId, newDocumentId);
+      // Update XPath expressions in the item
+      if (child instanceof ExpressionItem) {
+        child.expression = child.expression.replace(new RegExp(`\\$${oldDocumentId}\\b`, 'g'), `$${newDocumentId}`);
```

Comment:
This might need a bit more work for JSON parameter. When a JSON schema is attached to the source parameter, while the parameter name remains as what user defined, the actual variable used in xpath expression will get `-x` suffix. For example if the parameter name is `Account`, it will appear as `$Account-x` in xpath.